library/network_connections.py

```diff
@@ -1025,9 +1035,20 @@ def connection_create(self, connections, idx, connection_current=None):
                 s_ip6.set_property(
                     NM.SETTING_IP_CONFIG_ROUTE_METRIC, ip["route_metric6"]
                 )
-            for d in ip["dns"]:
-                if d["family"] == socket.AF_INET6:
-                    s_ip6.add_dns(d["address"])
+            for nameserver in ip["dns"]:
+                if nameserver["family"] != socket.AF_INET6:
+                    continue
+                if ipv6_is_disabled:
+                    raise MyError(
+                        "IPv6 needs to be enabled to support IPv6 nameservers."
+                    )
```

Comment:
This might be better checked in validate_post, @thom311 what do you think? I believe to remember that you might have suggested this in the past, too.

Reply:
good point. Errors should be caught as early as possible, and if a constraint only depends on settings from the input YAML, then such misconfiguration should already be caught when validating/normalizing the YAML. Tha's what `ArgValidator` does. In the first step, its validation only depends on the input YAML alone.

Reply:
I added the following into function `validate_connection_one()`: 

```
        if connection["ip"]["dns_options"] or connection["ip"]["dns_search"]:
            addrs4 = list(
                [
                    address
                    for address in connection["ip"]["address"]
                    if address["family"] == socket.AF_INET
                ]
            )
            if not connection["ip"]["dhcp4"] and not addrs4:
                raise ValidationError.from_connection(
                    idx,
                    "IPv4 needs to be enabled to support IPv4 nameservers.",
                )
            if connection["ip"]["ipv6_disabled"]:
                raise ValidationError.from_connection(
                    idx,
                    "IPv6 needs to be enabled to support IPv6 nameservers.",
                )
```